InteractionBase/src/vis/ViewSwitcherMenu.cpp

```diff
@@ -96,13 +118,19 @@ void ViewSwitcherMenu::endFocusMode(Visualization::Item *target)
 
 bool ViewSwitcherMenu::executeEntry(Visualization::Item* item)
 {
+	//static int nameCounter = 0;
 	if (inEditMode_) return false;
 	if (auto entry = DCast<VViewSwitcherEntry>(item))
 	{
 		QPoint pos = indexOf(item);
 		auto view = scene()->viewItems()->viewItem(entry->nameField()->text());
 		if (!view)
-			view = scene()->viewItems()->newViewItem(entry->nameField()->text(), pos);
+		{
+			auto name = entry->nameField()->text();
+			if (name == "Empty slot")
+				name = "View" + QString::number(pos.x()) + QString::number(pos.y()) + "";//QString::number(nameCounter++);
```

Comment:
So I made some changes, adding a static method isValidName(..) in the ViewItem class, which ensures that the specific name "Empty Slot" can't be used and that existing names will never be used.